src/foam/nanos/crunch/predicate/CapabilityDataUpdated.js

```diff
@@ -0,0 +1,40 @@
+/**
+ * @license
+ * Copyright 2021 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.nanos.crunch.predicate',
+  name: 'CapabilityDataUpdated',
+
+  extends: 'foam.mlang.predicate.AbstractPredicate',
+  implements: ['foam.core.Serializable'],
+
+  documentation: `Returns true if the data of capability was updated`,
+
+  javaImports: [
+    'foam.core.X',
+    'foam.nanos.crunch.Capability',
+    'foam.nanos.crunch.CapabilityJunctionStatus',
+    'foam.nanos.crunch.UserCapabilityJunction'
+  ],
+
+  methods: [
+    {
+      name: 'f',
+      javaCode: `
+        if ( ! ( obj instanceof X ) ) return false;
```

Comment:
When does this happen?

Reply:
I don't know, was using another code as an example
<img width="782" alt="Screen Shot 2021-09-29 at 3 20 20 PM" src="https://user-images.githubusercontent.com/58699954/135334612-ee01f28b-c412-4a7b-a603-731239fc596c.png">

Reply:
there is no shortage of people who have done this ... me, patrick, mayowa, chanman, eric ... list goes on ... I also just copied.
It was probably just a precaution at some point cause X is cast next.